doc/COMPONENTS-INTERACTION.md

```diff
@@ -0,0 +1,181 @@
+  * Connects with the Janus instance
+  * Fetches available rooms list
+  * Creates a new `Room` object for each room in fetched rooms list
+  * Creates a new `FeedConnection` object right after attaching to the plugin (`janus.plugin.videoroom`) (for entering a room)
+  * Subscribe to -
+    + existing feeds on `"joined"` event
+    + new feeds on `"event"` event
+  * Calls `ActionService` methods to 
+    + Enter/Leave a room
+    + Publish/Unpublish a feed/screen
+    + Start/Stop ignoring a feed
+    + Toggle a channel
+    + Remote join
+    + Disable/Enable audio/video for main feed
+  * Calls `FeedsService`methods to
+    + find `Feed` object using id
+    + find `"main" Feed` object
+  * Calls `DataChannelService` methods to
+    + Broadcast status information of all feeds when a data channel is established
+    + Recieve messages from a feed
+
+02. [room.factory.js](../src/app/components/room/room.factory.js) - It defines `Room` object containing data about video chat room and attributes from *janus.plugin.videoroom.cfg*. Some of its attributes are -
+  * id
+  * description
+  * max-publishers
+  * pin-required
+ etc.
+
+03. [feed-connection.factory.js](../src/app/components/feed/feed-connection.factory.js) - It defines `FeedConnection` object that manages the connection of a feed to the Janus server. Some of the attributes provided are - 
+  * `pluginHandle` object
+  * feed role (publisher/subscriber)
+  * `ConnectionConfig` object (using *connection-config.factory.js*)
+It also performs some important actions like - 
+  * sending a `join` request (`register()` and `listen()` methods)
+  * negotiating a WebRTC connection by -
+    + creating a WebRTC `offer` for sharing the audio and video with the Janus server (`publish()` method)
+    + creating a WebRTC `answer` for subscribing to a feed from the Janus server (`subscribe()` method)
+
+04. [action.service.js](../src/app/components/room/action.service.js) - Provides services by performing actions, like - 
+  * Creating new feed when
+    + entering a room
+    + publishing screen
+    + remote join
+  * Destroying feed when
+    + leaving a room
+    + unpublishing screen
+  * Disabling or enabling audio or video for the main feed
+  * Ignoring or stop ignoring a feed
+  * Toggling channel (muting or unmuting)
+  * Adding log entries for above actions (by creating `LogEntry` object defined in *log-entries.factory.js*)
+  * Sending chat text-message using `DataChannelService` (defined in *data-channel.service.js*) 
+
+05. [feeds.factory.js](../src/app/components/feed/feeds.factory.js) - Defines `Feed` object representing a Janus feed. It maintains data related to a feed for local representation, and, sending to remote peers. It contains attributes like -
+  * `FeedConnection` object
+  * `id` and `display` of feed
+  * other feed info -`isPublisher`, `isLocalScreen`, `isIgnored`
+ It also contains methods that perform actions like - 
+  * changing the display
```

Comment:
Clarify that "the display" means the name in this context. Is probably not obvious to a newcomer.